### ISSUE TITLE
ML-Benchmark prediction submission: GNN4CD_Rall

### ISSUE THREAD

**Issue by Sera91** [WCRP-CORDEX/ml-benchmark#34]
### Institution ID

ICTP

### Institution Name

Abdus Salam International Center for Theoretical Physics

### Emulator identifier (`emulator_id`)

GNN4CD_Rall_noRNN-orog

### Emulator Description

The emulator used is a GNN based architecture, developed using Pytorch, corresponding to the Rall model version described in the publication "Blasone et al., 2025" (Blasone V, Coppola E, Sanguinetti G, Arora V, Di Gioia S, Bortolussi L. Graph neural networks for hourly precipitation projections at the convection permitting scale with a novel hybrid imperfect framework. Environmental Data Science. 2025;4:e47. doi:10.1017/eds.2025.10022 ).
The model contains 10 learnable layers: 1 dense, 1 GraphConv, 5 GATv2Conv, and 3 fully connected layers.
The only difference with respect to the published version is that we remove the first RNN layer.
We applied a z-score method to standardize the input predictors, while
We do min-max scaling for the temperature and log1p transformation for the precipitation target.
As predictors at low resolution we use all the fields () and the 3 pressure levels provided, while as predictors at high-resolution we use the orography and the lat,lon at high-resolution.

We perform a separate supervised training of the model for the precipitation and the temperature, using the same training strategy, with the same optimizer (Adam) but different losses and initial learning rate.

For the training for precipitation downscaling we have:
LR_0 = 0.0008
Alpha=0.01
Scheduler =CosineAnnealingLR
Loss function: MSE+alpha*quantized loss

For temperature downscaling we have
LR_0 = 0.001
Scheduler =CosineAnnealingLR
Loss function: MSE

### Hardware and Training Details

We trained the model on one node of the Booster partition of Leonardo cluster (Leonardo documentation), containing 4 NVIDIA A100 GPUs,
measuring a training time of 60 seconds per epoch.
For inference we use the same setup and we measure 45-50 seconds of inference time per year of prediction. 

### Stochastic/Probabilistic Output

no

### Reference URL

doi:10.1017/eds.2025.10022

### Repository for Reproducibility

https://github.com/valebl/GNN4CD
